src/emdrv/gpioint.rs

```diff
@@ -1,19 +1,111 @@
-pub type IrqCallback = extern fn (pin: u8);
+use cmsis::nvic;
+use {gpio, int};
+use core::option::Option;
+use core::option::Option::{Some, None};
 
-extern {
-    pub fn GPIOINT_Init();
-    pub fn GPIOINT_CallbackRegister(pin: u8, callback: IrqCallback);
-    pub fn STATIC_INLINE_GPIOINT_CallbackUnRegister(pin: u8);
-}
+pub type IrqCallback = fn (u8);
+
+static mut CALLBACKS: [Option<IrqCallback>; 16] = [
+    None,None,None,None,
+    None,None,None,None,
+    None,None,None,None,
+    None,None,None,None
+];
 
 pub fn init() {
-    unsafe { GPIOINT_Init() }
+    nvic::clear_pending_irq(nvic::IRQn::GPIO_ODD);
+    nvic::enable_irq(nvic::IRQn::GPIO_ODD);
+    nvic::clear_pending_irq(nvic::IRQn::GPIO_EVEN);
+    nvic::enable_irq(nvic::IRQn::GPIO_EVEN);
+}
+
+pub fn set(pin: u8, callback: Option<IrqCallback>) {
+
+    int::disable();
+    unsafe { CALLBACKS[pin as usize] = callback; }
+    int::enable();
+}
+
+pub fn get(pin: u32) -> Option<IrqCallback> {
+
+    int::disable();
+    let callback = unsafe { CALLBACKS[pin as usize] };
+    int::enable();
+
+    callback
+}
+
+pub fn register(pin: u8, callback: IrqCallback) {
```

Comment:
Don't know if `register`/ `unregister` are redundant or not, since the `set` function is public anyways... 
Make the set-function private or remove register/unregister, or keep both as is?